api/v1alpha6/openstackcluster_conversion.go

```diff
@@ -546,6 +546,29 @@ func restorev1beta1BastionStatus(previous *infrav1.BastionStatus, dst *infrav1.B
 	dst.Resources = previous.Resources
 }
 
+/* Condition status */
```

Comment:
Maybe we don't need that?
The way I made it work locally is to add this to `restorev1beta1ClusterStatus`:

```golang
if previous.Conditions != nil {
    dst.Conditions = previous.Conditions
}
```